Tools/paletteable sprite editor/js/marvinj-0.9.js

```diff
@@ -0,0 +1,2665 @@
+						
+						nx = x + (j-xC);
+						ny = y + (i-yC);
+						
+						if(nx > 0 && nx < imgOut.getWidth() && ny > 0 && ny < imgOut.getHeight()){
+							imgOut.setBinaryColor(nx, ny, true);
+						}
+					}
+				}
+			}
+		}
+	}
+
+
+	function Erosion(){
+		MarvinAbstractImagePlugin.super(this);
+		this.load();
+	}
+	
+	Erosion.prototype.load = function(){
+		
+		this.matrix = MarvinJSUtils.createMatrix2D(3,3,true);
+		this.setAttribute("matrix",3);
+	}
+	
+	Erosion.prototype.process = function
+	(
+		imgIn, 
+		imgOut,
+		attributesOut,
+		mask, 
+		previewMode
+	)
+	{
+		var matrix = this.getAttribute("matrix");
+		
+		if(imgIn.getColorModel() == MarvinImage.COLOR_MODEL_BINARY && matrix != null){
+			
+			MarvinImage.copyColorArray(imgIn, imgOut);
+			
+			for(var y=0; y<imgIn.getHeight(); y++){
+				for(var x=0; x<imgIn.getWidth(); x++){
+					this.applyMatrix(x, y, matrix, imgIn, imgOut);
+				}
+			}
+		}
+	}
+	
+	Erosion.prototype.applyMatrix = function
+	(
+		x,
+		y,
+		matrix,
+		imgIn,
+		imgOut
+	){
+		
+		var nx,ny;
+		
+		var xC=Math.floor(matrix[0].length/2);
+		var yC=Math.floor(matrix.length/2);
+		
+		if(!imgIn.getBinaryColor(x, y)){
+			for(var i=0; i<matrix.length; i++){
+				for(var j=0; j<matrix[0].length; j++){
+					
+					if((i != yC || j != xC) && matrix[i][j]){
+						
+						nx = x + (j-xC);
+						ny = y + (i-yC);
+						
+						if(nx >= 0 && nx < imgOut.getWidth() && ny >= 0 && ny < imgOut.getHeight()){
+							imgOut.setBinaryColor(nx, ny, false);
+						}
+					}
+				}
+			}
+		}
+	}
+
+
+	function IteratedFunctionSystem(){
+		MarvinAbstractImagePlugin.super(this);
+		this.load();
+	}
+	
+	IteratedFunctionSystem.prototype.load = function(){
+		this.rules = [];
+		this.EXAMPLE_RULES = "0,0,0,0.16,0,0,0.01\n"+
+											"0.85,0.04,-0.04,0.85,0,1.6,0.85\n"+
+											"0.2,-0.26,0.23,0.22,0,1.6,0.07\n"+
+											"-0.15,0.28,0.26,0.24,0,0.44,0.07\n";
+											
+		this.setAttribute("rules", this.EXAMPLE_RULES);
+		this.setAttribute("iterations", 1000000);
+	};
+
+		/*
+	private MarvinAttributesPanel	attributesPanel;
+	private MarvinAttributes 		attributes;
+	
+	private List<Rule> rules;
+	
+	// Testing String
+	private final static String EXAMPLE_RULES = 	"0,0,0,0.16,0,0,0.01\n"+
+											"0.85,0.04,-0.04,0.85,0,1.6,0.85\n"+
+											"0.2,-0.26,0.23,0.22,0,1.6,0.07\n"+
+											"-0.15,0.28,0.26,0.24,0,0.44,0.07\n";
+	@Override
+	public void load() {
+		attributes = getAttributes();
+		attributes.set("rules", EXAMPLE_RULES);
+		attributes.set("iterations", 1000000);
+		
+		rules = new ArrayList<Rule>();
+	}
+	*/
+
+	IteratedFunctionSystem.prototype.process = function
+	(
+		imageIn, 
+		imageOut,
+		attributesOut,
+		mask, 
+		previewMode
+	)
+	{
+		this.loadRules();
+		var iterations = this.getAttribute("iterations");
+		
+		var x0 = 0;
+		var y0 = 0;
+		var x,y;
+		var startX;
+		var startY;
+		var factor;
+		
+		var minX=999999999,minY=999999999,maxX=-999999999,maxY=-99999999;
```

Comment:
Codacy found an issue: [The numeric literal '99999999' will have at different value at runtime.](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6395467)